app/portainer/views/endpoints/create/createEndpointController.js

```diff
@@ -84,6 +85,12 @@ function ($q, $scope, $state, $filter, clipboard, EndpointService, GroupService,
     createAzureEndpoint(name, applicationId, tenantId, authenticationKey, groupId, tagIds);
   };
 
+  $scope.onCreateTag = async function onCreateTag(tagName) {
+    const tag = await TagService.createTag(tagName);
+    $scope.availableTags = $scope.availableTags.concat(tag);
+    $scope.formValues.TagIds = $scope.formValues.TagIds.concat(tag.Id);
+  }
+
```

Comment:
Wrap with non async function. The main goal for this wrapper is to trigger a digest cycle when the async function returns. See https://github.com/portainer/portainer/pull/2945 for more details.

Also, protect the await with a try/catch closure and show a notification if error occurs.